src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,315 @@
+    # Load Devices located within at least one exclusion zones
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b_exz.json')))
+    # Load Devices located within 50 meters of at least one exclusion zones
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c_exz.json')))
+   
+    # Prepare 3 grant requests.
+    # 1. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3625000000.0,
+        'highFrequency': 3635000000.0
+    }
+
+    # 2. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_1['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3625000000.0
+    grant_1['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+
+    # 3. With frequency range partially overlapping with the frequency 
+    #     range of the nearest exclusion zone.
+    grant_2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_2['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3625000000.0
+    grant_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
+    grants = [grant_0, grant_1, grant_2]
+    config = {
+        'fccIds': [(d['fccId'], 47) for d in devices],
+        'userIds': [d['userId'] for d in devices],
+        'registrationRequests': devices,
+        'expectedResponseCodes': [(0,), (0,), (0,)],
+        'grantRequests' : grants,
+        'expectedGrantResponseCodes': [(0,), (400,), (400,)],
+    }
+    writeConfig(filename, config)
+
+
+
+  @configurable_testcase(generate_EXZ_1_default_config)
+  def test_WINNF_FT_S_EXZ_1(self,config_filename):
+    """Injected Exclusion Zones
+
+    The response should be SUCCESS for CBSDs located outside 50meters of all 
+    Exclusion Zones.
+    The responseCode = 400 for CBSDs located within 50meters of all Exclusion Zones 
+    or inside Exclusion Zones. 
+    """
+
+    #Load Exclusion Zones
+    exz_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'exz_record_0.json')))
+
+    # Inject EXZ database records
+    # TODO Need to figure out how to send freqRange
+    """
+    zone_id = self._sas_admin.InjectExZone({'record':exz_record}, \
+                     {'frequencyRange':{'lowFrequency': 3620000000.0,\
+                      'highFrequency': 3630000000.0}})
+    """
+    self._sas_admin.InjectExZone({'record': exz_record})
```

Comment:
The exclusion zones and corresponding frequency ranges should be in the configuration itself. Please note that this test case needs to support multiple exclusion zones simultaneously.

Reply:
Creation and validation with 2 exclusion zones is added to the script. 

WINNF-TS-0096-V1.2.0-SAS-SAS Protocol Technical Specification-11-20-2017 - Document doesn’t provide frequency range configuration object. Currently not added in the exz_record